### PR TITLE
fix: root workspace path containment

### LINKED ISSUE #6: bug: path containment check fails at filesystem root
## Summary

The untracked file path containment check in `collectWorkingTreeContext()` incorrectly rejects all files when the workspace root is at the filesystem root (`/` on Unix, `C:\` on Windows).

## Location

`plugins/gemini/scripts/lib/git.mjs:186`

```javascript
const realCwd = fs.realpathSync(cwd);
if (!realPath.startsWith(realCwd + path.sep) && realPath !== realCwd) {
  results.push({ path: file, skipped: "outside workspace" });
}
```

## Problem

When the workspace is at a filesystem root:

| Platform | `realCwd` | `realCwd + path.sep` | Result |
|----------|-----------|----------------------|--------|
| Unix | `/` | `//` | `/file.txt` does NOT start with `//` — incorrectly rejected |
| Windows | `C:\` | `C:\\` | `C:\file.txt` does NOT start with `C:\\` — incorrectly rejected |

Every untracked file will be marked as `"outside workspace"` and skipped from the review context.

## Suggested Fix

Guard against roots that already end with the separator:

```javascript
const realCwd = fs.realpathSync(cwd);
const prefix = realCwd.endsWith(path.sep) ? realCwd : realCwd + path.sep;
if (!realPath.startsWith(prefix) && realPath !== realCwd) {
  results.push({ path: file, skipped: "outside workspace" });
}
```

## Impact

- **Severity:** Low
- Extremely unlikely in practice — no one runs a workspace at `/` or `C:\`
- If triggered, the symptom is silent: untracked files are skipped with `"outside workspace"` but no error is raised to the user
- Could theoretically occur in containerized environments where the app root is `/`

### PR DESCRIPTION
**Pull request by sakibsadmanshajib**
## Summary

- Fix untracked-file containment checks when the workspace realpath is a filesystem root.
- Add a regression test that simulates a root-resolved workspace and verifies untracked file contents are still included.

## Root Cause

`readUntrackedFiles()` built its containment prefix as `realCwd + path.sep`. At filesystem roots, `realCwd` already ends with the separator, so the prefix became `//` on Unix or `C:\\` on Windows and valid root-level files failed `startsWith()`.

## Test Plan

- `node --test tests/git.test.mjs`
- `npm test`

Fixes #6

## Release Notes

* **Bug Fixes**
  * Improved validation for correctly identifying files outside workspace boundaries.

* **Tests**
  * Added test coverage for untracked file handling in edge cases.

### CODE REVIEW

File: plugins/gemini/scripts/lib/git.mjs [hunk 1]
```diff
@@ -183,7 +183,8 @@ export function readUntrackedFiles(cwd, files, options = {}) {
       // Verify the resolved path stays inside the workspace.
       const realPath = fs.realpathSync(fullPath);
       const realCwd = fs.realpathSync(cwd);
-      if (!realPath.startsWith(realCwd + path.sep) && realPath !== realCwd) {
+      const workspacePrefix = realCwd.endsWith(path.sep) ? realCwd : realCwd + path.sep;
```
Comment (sakibsadmanshajib) [inline plugins/gemini/scripts/lib/git.mjs]: **Minor / Design suggestion (all 3 reviewers):** Consider replacing the `startsWith` + separator-arithmetic check with `path.relative`:

```js
const rel = path.relative(realCwd, realPath);
const outside = rel.startsWith("..") || path.isAbsolute(rel);
if (outside) {
  results.push({ path: file, skipped: "outside workspace" });
  continue;
}
```

This avoids the separator edge case that caused this bug entirely, handles the `realPath === realCwd` case (returns `""`), and reads as a security assertion rather than a string prefix dance. The current fix is correct — this is a long-term idiom suggestion, not a blocker.

File: plugins/gemini/scripts/lib/git.mjs [hunk 2]
```diff
@@ -183,7 +183,8 @@ export function readUntrackedFiles(cwd, files, options = {}) {
...
+      if (!realPath.startsWith(workspacePrefix) && realPath !== realCwd) {
```
Comment (sakibsadmanshajib) [inline plugins/gemini/scripts/lib/git.mjs]: Confirmed correct across platforms:
- POSIX root `realCwd === "/"` → `endsWith("/")` is true → `workspacePrefix = "/"` (no double slash). ✅
- Windows drive root `realCwd === "C:\\"` → `endsWith("\\")` is true on Win32 where `path.sep === "\\"` → no doubling. ✅
- Normal case `realCwd === "/home/user/proj"` → appends separator as before. ✅

The `realPath !== realCwd` branch is still reachable and meaningful (case where the listed file resolves to the workspace root itself).

File: tests/git.test.mjs [hunk 3]
```diff
@@ -84,6 +84,41 @@ test("collectReviewContext skips broken untracked symlinks instead of crashing",
...
+  run("git", ["add", "app.js"], { cwd });
+  run("git", ["commit", "-m", "init"], { cwd });
+
+  const file = "notes.txt";
+  const fullPath = path.join(cwd, file);
+  fs.writeFileSync(fullPath, "review me\n");
+
+  const root = path.parse(cwd).root;
+  const originalRealpathSync = fs.realpathSync;
+  fs.realpathSync = (target, options) => {
```
Comment (sakibsadmanshajib) [inline tests/git.test.mjs]: **Important — monkey-patch fragility (ECC + Gemini):** This mutates the `fs` module object directly. It works today because `git.mjs:5` uses `import fs from "node:fs"` (default import — same object reference). If that import is ever refactored to a named import like `import { realpathSync } from "node:fs"`, the binding is captured at module load, this monkey-patch silently stops intercepting, and the test passes vacuously with a false sense of security.

Suggest either:
- Adding a comment here documenting this coupling, or
- Refactoring `readUntrackedFiles` to accept an injectable `realpath` dependency so the test can pass a fake directly.

File: tests/git.test.mjs [hunk 4]
```diff
@@ -84,6 +84,41 @@ test("collectReviewContext skips broken untracked symlinks instead of crashing",
...
+  try {
+    const result = collectReviewContext(cwd, { scope: "working-tree" });
+
+    assert.deepEqual(result.context.untrackedContents, [
+      { path: file, content: "review me\n" }
+    ]);
+  } finally {
+    fs.realpathSync = originalRealpathSync;
+  }
+});
```
Comment (sakibsadmanshajib) [inline tests/git.test.mjs:150]: **Important — missing negative test (all 3 reviewers):** This test proves the *inclusion* path works when `cwd` resolves to `/`, but does NOT prove the containment guard still *rejects* a traversal attempt under the same scenario. A regression that silently removed `startsWith(workspacePrefix)` would pass CI.

Please add a companion negative case. Sketch:

```js
test("readUntrackedFiles rejects files outside root workspace", () => {
  const cwd = makeTempDir();
  initGitRepo(cwd);
  fs.writeFileSync(path.join(cwd, "escape.txt"), "nope\n");
  run("git", ["add", "."], { cwd });
  run("git", ["commit", "-m", "init"], { cwd });
  fs.writeFileSync(path.join(cwd, "legit.txt"), "ok\n");

  const originalRealpathSync = fs.realpathSync;
  fs.realpathSync = (target) => {
    const t = path.resolve(String(target));
    if (t === path.resolve(cwd)) return "/";
    // escape: realpath returns a path outside the mocked root workspace
    return "/etc/passwd";
  };

  try {
    const result = collectReviewContext(cwd, { scope: "working-tree" });
    assert.ok(result.context.untrackedContents.some(
      (u) => u.skipped === "outside workspace"
    ));
  } finally {
    fs.realpathSync = originalRealpathSync;
  }
});
```

This locks in the security property — without it, the positive test alone is not evidence that the guard still works.

**Review by sakibsadmanshajib — COMMENTED** [review]
Multi-reviewer analysis: **Superpowers** (proceed), **ECC typescript-reviewer** (ship with caution), **Gemini 2.5 Pro adversarial** (patches bug, recommends refactor). No CRITICAL issues — fix is correct for POSIX `/` and Windows `C:\`, and grep confirmed no other `+ path.sep` call sites.

Two Important findings all reviewers converge on:
1. Test lacks a negative case (traversal rejection under root cwd)
2. `path.relative()` would be a more robust idiom

See inline comments for specifics.

File: plugins/gemini/scripts/lib/git.mjs (hunk 1 above)
Comment (sakibsadmanshajib) [reply plugins/gemini/scripts/lib/git.mjs]: Addressed in e9598bdab7fa3413930e43ddd077bdf3e055e427. The containment check now uses `path.relative` with `rel === ".." || rel.startsWith(".." + path.sep) || path.isAbsolute(rel)` so root workspaces avoid separator arithmetic without rejecting in-workspace names like `..foo`. Verification: `npm test` (44/44) and `git diff --check`.

File: plugins/gemini/scripts/lib/git.mjs (hunk 2 above)
Comment (sakibsadmanshajib) [reply plugins/gemini/scripts/lib/git.mjs]: Superseded by e9598bdab7fa3413930e43ddd077bdf3e055e427. The root-specific `workspacePrefix` branch is gone; containment is now expressed via `path.relative`, covering POSIX root, drive roots, and same-path cases without separator arithmetic. Verification: `npm test` (44/44) and `git diff --check`.

File: tests/git.test.mjs (hunk 3 above)
Comment (sakibsadmanshajib) [reply tests/git.test.mjs]: Addressed in e9598bdab7fa3413930e43ddd077bdf3e055e427. `readUntrackedFiles` now accepts an injectable `realpathSync`, and the tests pass a fake directly instead of monkey-patching the `fs` module object. Verification: `npm test` (44/44) and `git diff --check`.

File: tests/git.test.mjs (hunk 4 above)
Comment (sakibsadmanshajib) [reply tests/git.test.mjs:150]: Addressed in e9598bdab7fa3413930e43ddd077bdf3e055e427. Added a companion negative test that simulates an untracked file resolving outside the workspace and asserts it is skipped as `outside workspace`. Verification: `npm test` (44/44) and `git diff --check`.
